pyccel/ast/core.py

```diff
@@ -177,7 +177,75 @@ def __init__(self, var):
         # Call the base class constructor with the parameters it needs
         super(AstFunctionResultError, self).__init__(msg)
 
+# Pow, Add, Mul need to inherite sympy.Boolean to be able to use them in a logical expression
 
+class Pow(sp_Pow, sp_Boolean, PyccelAstNode):
+    def __new__(cls, *args, evaluate = False, **kwargs):
+        return sp_Pow.__new__(cls, *args, evaluate = evaluate, **kwargs)
+
+    def _eval_subs(self, old, new):
+        args = self.args
+        args_ = [self.base._subs(old, new),self.exp._subs(old, new)]
+        args  = [args_[i] if args_[i] else args[i] for i in range(len(args))]
+        expr = Pow(args[0], args[1], evaluate=False)
+        return expr
+
+    def _eval_evalf(self,prec):
+        return sp_Pow(self.base,self.exp).evalf(prec)
+
+    @property
+    def rank(self):
```

Comment:
Given that `Pow` inherits the `rank` property from `PyccelAstNode`, I cannot see a good reason to override it here. It seems more natural to me to compute the correct value of `self._rank` when we initialize a `Pow` object in the `__init__` function, which is called after `__new__`:
```
 def __init__(cls, *args, **kwargs):
    self._rank = max(getattr(a,'rank',0) for a in self._args)
    # TODO: Use broadcasting rules to compute self._shape
```
@EmilyBourne, @saidctb: What do you think?

Reply:
I agree

Reply:
I agree